Update on-call engineer handbook entries (#38641)
fleetdm/fleet: handbook/company/product-groups.md

#### Awaiting QA

Bugs will be verified as fixed by QA when they are placed in the "Awaiting QA" column of the relevant product group's sprint board. If the bug is verified as fixed, it is moved to the "Ready for release" column of the sprint board. Otherwise, the remaining issues are noted in a comment, and it is moved back to the "In progress" column of the sprint board.

## Engineering on-call

Fleet maintains two engineering on-call rotations: one for business-hours escalation support, and one for after-hours response to critical production incidents.

### Business hours on-call engineer

Customer Success members are instructed to `@oncall` in `#help-engineering` to contact the on-call engineer for support escalations or product incidents occurring during business hours.

In some cases, the Customer Success team member will continue to communicate with the requestor. In others, the on-call engineer will communicate directly. Team members should use their judgment and discuss on a case-by-case basis how to best communicate with community members and customers.

> The current business-hours on-call rotation is reflected in the [📈 KPIs spreadsheet (confidential Google sheet)](https://docs.google.com/spreadsheets/d/1Hso0LxqwrRVINCyW_n436bNHmoqhoLhC8bcbvLPOs9A/edit#gid=0&range=F2).

Reviewer: ```suggestion
The current business-hours on-call rotation is reflected in the [📈 KPIs spreadsheet (confidential Google sheet)](https://docs.google.com/spreadsheets/d/1Hso0LxqwrRVINCyW_n436bNHmoqhoLhC8bcbvLPOs9A/edit?gid=1831267053#gid=1831267053).
```

2. Create a new thread in `#help-engineering`, tagging the relevant product group and including any findings or links from your investigation.

#### Changing of the guard

The business-hours on-call engineer changes each week on Monday. A Slack reminder notifies the `@oncall` alias of the handoff. Please do the following:

1. The new on-call engineer should update the `@oncall` Slack user group to point to themselves.
2. Hand off active conversations (Slack threads, issues, PRs, etc.).  
   - Share relevant threads with the new on-call engineer  
   - Subscribe them to notifications  
   - Unsubscribe the former on-call engineer as appropriate  

Anything not clearly handed off remains the responsibility of the former on-call engineer.

In the Slack reminder thread, the outgoing on-call engineer should include a brief retrospective:

1. Most common support requests that week  
2. Documentation page reviewed and changes made  
3. How the rest of the on-call time was spent  

### After-hours incident on-call engineer

> Customer Success members are instructed to `@incident-oncall` in `#help-engineering` to report [P0 incidents](https://fleetdm.com/handbook/company/communications#high-priority-user-stories-and-bugs) occurring after hours.

Reviewer 2: There were more than that, but once the fire was out the tickets were de-escalated. Looking at postmortems would give you a better read on how many P0s we actually had.

> Customer Success members are instructed to `@incident-oncall` in `#help-engineering` to report [P0 incidents](https://fleetdm.com/handbook/company/communications#high-priority-user-stories-and-bugs) occurring after hours.

Author: ```suggestion
Customer Success members are instructed to `@incident-oncall` in `#help-engineering` to report [P0 incidents](https://fleetdm.com/handbook/company/communications#high-priority-user-stories-and-bugs) occurring after business hours.
```

> Incidents that occur during business hours should be handled by the business-hours on-call engineer and should not use `@incident-oncall`.

Author: I'm open to doing that, but the net result is that @incident-oncall becomes 24 hours. It's certainly cleaner for the business to know "@oncall is for support escalations during business hours, @incident-oncall is for any P0 incidents". 

I think it's ultimately up to the folks on after-hours incident on-call if they're willing to cover 24 hours/day.

> Incidents that occur during business hours should be handled by the business-hours on-call engineer and should not use `@incident-oncall`.

Author: Yep agree this could be smoothed out. I think we proceed as-is but open to changing with input from folks in the rotation.

> In some cases, the Customer Success team member will continue communicating with the impacted customer or customers. In others, the on-call engineer will communicate directly. Team members should use their judgment and coordinate on a case-by-case basis.

Reviewer 2: Probably useful to be explicit here that whoever triggers the incident is the point person for comms unless that person explicitly hands off to someone else.

This works as long as infrastructure can't create incidents on its own, but we can cross that bridge when we come to it.

> In some cases, the Customer Success team member will continue communicating with the impacted customer or customers. In others, the on-call engineer will communicate directly. Team members should use their judgment and coordinate on a case-by-case basis.

Author: ```suggestion
The Customer Success team member who reports the incident will continue communicating with the impacted customer or customers and is responsible for maintaining open lines of communication during the incident. If needed, the on-call engineer may be asked to communicate directly with impacted customers to facilitate faster resolution. Team members should use their judgment and coordinate on a case-by-case basis.
```

Each product group maintains two engineers assigned to after-hours incident on-call. Engineers in this rotation should be comfortable leading mitigation efforts during an after-hours production incident.

> The after-hours incident on-call rotation and handoff are handled automatically by incident.io. The on-call rotation may be adjusted in incident.io with approval from the EMs of any product groups affected. Any changes should be made before the start of the sprint so that capacity can be planned accordingly.

#### After-hours incident on-call responsibilities

> The after-hours incident on-call engineer is responsible for responding within one hour to [P0 incidents](https://fleetdm.com/handbook/company/communications#high-priority-user-stories-and-bugs) occurring outside of business hours and triaging into our [incident response process](https://fleetdm.com/handbook/company/communications#incident-response-process). 

Author: ```suggestion
The after-hours incident on-call engineer is responsible for responding within one hour to [P0 incidents](https://fleetdm.com/handbook/company/communications#high-priority-user-stories-and-bugs) occurring outside of business hours and triaging into our [incident response process](https://fleetdm.com/handbook/company/engineering#incident-response-process). 
```
